### PR TITLE
Proposition of new PR template

### PR DESCRIPTION
**Pull request by farmaazon**
[ci no changelog needed]

### Pull Request Description

After an incident with IDE initialization (see https://github.com/enso-org/enso/pull/3274) we discussed briefly a need for adding new checks to the PR. Here goes my proposition.

### Checklist

Please include the following checklist in your PR:

- ~~[ ] The documentation has been updated if necessary.~~
- ~~[ ] All code conforms to the [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md), [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md), and [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guides.~~
- [x] All documentation and configuration conforms to the [markdown](https://github.com/enso-org/enso/blob/develop/docs/style-guide/markdown.md) and [YAML](https://github.com/enso-org/enso/blob/develop/docs/style-guide/yaml.md) style guides.
- ~~[ ] All code has been tested where possible.~~

### CODE REVIEW

**Review by vitvakatu — APPROVED** [review]

**Review by kazcw — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -18,5 +18,6 @@ Please include the following checklist in your PR:
 
 - [ ] The documentation has been updated if necessary.
 - [ ] All code conforms to the [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md), [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md), and [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guides.
-- [ ] All documentation and configuration conforms to the [markdown](https://github.com/enso-org/enso/blob/develop/docs/style-guide/markdown.md) and [YAML](https://github.com/enso-org/enso/blob/develop/docs/style-guide/yaml.md) style guides.
-- [ ] All code has been tested where possible.
+- All code has been tested:
+  - [ ] Unit tests have been written where possible.
+  - [ ] If GUI codebase was changed: BOTH `./run dist` and `./run watch` have been tested.
```
Comment (akavel) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
  - [ ] If GUI codebase was changed: GUI was tested when built using BOTH `./run dist` and `./run watch`.
```

I would clarify that what we want is to have *the GUI* tested, not just that the `./run watch` and `./run dist` commands succeeded to compile the PR.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -18,5 +18,6 @@ Please include the following checklist in your PR:
 
 - [ ] The documentation has been updated if necessary.
 - [ ] All code conforms to the [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md), [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md), and [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guides.
-- [ ] All documentation and configuration conforms to the [markdown](https://github.com/enso-org/enso/blob/develop/docs/style-guide/markdown.md) and [YAML](https://github.com/enso-org/enso/blob/develop/docs/style-guide/yaml.md) style guides.
```
Comment (jdunkerley) [inline .github/PULL_REQUEST_TEMPLATE.md:21]: Is this checkbox deliberately removed?

Comment (akavel) [reply .github/PULL_REQUEST_TEMPLATE.md:21]: yes:  ["those [are] already checked by CI"](https://discord.com/channels/401396655599124480/942778039262126090/942792057070694480)

**Review by jdunkerley — APPROVED** [review]

**Review by wdanilo — APPROVED** [review]
I'd give the same comment as @akavel did.

**Review by akavel — APPROVED** [review]

**Review by MichaelMauderer — APPROVED** [review]

**Pull request merged**